astroid/inference.py

```diff
@@ -58,9 +71,24 @@
 from astroid.interpreter import dunder_lookup
 from astroid.manager import AstroidManager
 
+if TYPE_CHECKING:
+    from astroid.objects import Property
+
 # Prevents circular imports
 objects = util.lazy_import("objects")
 
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
+else:
+    from typing_extensions import TypedDict
+
+T_FunctionDefSelf = TypeVar("T_FunctionDefSelf", bound=nodes.FunctionDef)
```

Comment:
```suggestion
FunctionDefT = TypeVar("FunctionDefT", bound=nodes.FunctionDef)
```
Let's use the "right" naming style. At some point I should probably look at that pylint PR again 🙈

Reply:
You'll be happy to know I split it up into two smaller, but still hellish, PRs 😄